File: app/lib/core/models/rbac.dart
// Espelha o catálogo RBAC do backend (api/app/services/rbac.rb).
// A API serializa permissions com atalhos: "resource:*" = todas as ações
// do resource, "*" = todas as ações de todos os resources (domínio + admin).
const String kPermissionWildcard = '*';

abstract class Resources {
  // Domínio (Rbac::DOMAIN_RESOURCES)
  static const String journey = 'journey';
  static const String history = 'history';
  static const String profile = 'profile';

  // Administração (Rbac::ADMIN_RESOURCES)
  static const String users = 'users';
  static const String roles = 'roles';
  static const String permissions = 'permissions';
  static const String notification = 'notification';
  static const String weeklyReview = 'weekly_review';
}

abstract class Actions {
  static const String view   = 'view';
  static const String create = 'create';
  static const String update = 'update';
  static const String delete = 'delete';
}

File: api/app/services/rbac.rb
# frozen_string_literal: true

# RBAC catalog structure, mirrored in the Flutter app (app/lib/core/models/rbac.dart).
# Domain resources are the ones the client app has user-facing screens for;
# admin resources gate administration features. Both groups are serialized
# on login/me so the app can gate admin UI.
module Rbac
  ADMIN_RESOURCES = %w[users roles permissions notification weekly_review].freeze
  DOMAIN_RESOURCES = %w[journey history profile].freeze
  ALL_RESOURCES = (ADMIN_RESOURCES + DOMAIN_RESOURCES).freeze
  ACTIONS = %w[view create update delete].freeze

  # Serialized shorthand: "resource:*" = every action on the resource,
  # "*" = every action on every resource (domain + admin).
  WILDCARD = "*"
end
